src/crafting.rs

```diff
@@ -0,0 +1,128 @@
+use colored::*;
+use std::fmt;
+
+#[derive(Debug)]
+pub struct Recipe {
+  pub id: &'static str,
+  pub name: &'static str,
+  pub description: &'static str,
+  pub items_needed: Vec<(&'static str, usize)>,
+  pub tools_needed: Vec<&'static str>,
+  pub upgrades_needed: Vec<&'static str>,
+  pub result: Vec<&'static str>,
+  pub category: RecipeCategory,
+}
+
+impl fmt::Display for Recipe {
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    write!(f, "{}", self.name.bold())
+  }
+}
+
+#[derive(Debug)]
+pub enum RecipeCategory {
+  Consumable,
+  Tool,
+  Weapon,
+  Other,
+}
+
+pub fn recipes() -> [Recipe; 8] {
```

Comment:
I am not sure but it looks like the `_needed` vectors are really tied to the `category` so maybe it's better to include these in the `RecipeCategory` variants? Do you need a `tool` to create another `tool`?